File: src/main/java/com/dozycoffee/wms/warehouse/application/port/in/ReleaseWorkAreaUseCase.java
package com.dozycoffee.wms.warehouse.application.port.in;

import com.dozycoffee.wms.warehouse.application.port.in.command.ReleaseWorkAreaCommand;
import com.dozycoffee.wms.warehouse.application.port.in.result.WorkAreaResult;
import reactor.core.publisher.Mono;

public interface ReleaseWorkAreaUseCase {

    Mono<WorkAreaResult> release(ReleaseWorkAreaCommand command);
}

File: src/main/kotlin/com/dozycoffee/wms/warehouse/application/port/in/ReleaseWorkAreaUseCase.kt
package com.dozycoffee.wms.warehouse.application.port.`in`

import com.dozycoffee.wms.warehouse.application.port.`in`.command.ReleaseWorkAreaCommand
import com.dozycoffee.wms.warehouse.application.port.`in`.result.WorkAreaResult
import reactor.core.publisher.Mono

interface ReleaseWorkAreaUseCase {
    fun release(command: ReleaseWorkAreaCommand): Mono<WorkAreaResult>
}
